app/assets/javascripts/discourse/app/components/d-modal.hbs

```diff
@@ -25,43 +25,44 @@
       <div class="modal-middle-container">
         <div class="modal-inner-container">
           {{yield to="aboveHeader"}}
+          {{#if this.showHeader}}
+            {{#if
+              (or
+                this.dismissable
+                @title
```

Comment:
Rather than wrapping the whole thing in another `if` I'd add it here:

```suggestion
                (and @title (not @hideHeader))
```

and would remove the `showHeader` getter

Reply:
Yeah agreed on combining it into a single `{{#if`, and using `@hideHeader` directly instead of a `this.showHeader` getter 👍 

But I don't think we should combine it into `dismissable`. The dismissable boolean is also used to prevent things like click-outside and 'esc'.

So it would be 

```hbs
{{#if
  (and
    (not @hideHeader)
    (or
      this.dismissable
      @title
      (has-block "headerBelowTitle")
      (has-block "headerAboveTitle")
    )
  )
}}
```

(honestly I'd love to put all this in a single getter... but we don't have access to `has-block` in javascript 😢)

Reply:
Im fine with this, just steered away at first because... IF, AND, NOT, OR all together like this is pretty confusing 😜